src/marked.js

```diff
@@ -108,28 +108,35 @@ export function marked(src, opt, callback) {
   function onError(e) {
     e.message += '\nPlease report this to https://github.com/markedjs/marked.';
     if (opt.silent) {
-      return '<p>An error occurred:</p><pre>'
+      const msg = '<p>An error occurred:</p><pre>'
         + escape(e.message + '', true)
         + '</pre>';
+      if (opt.async) {
+        return Promise.reject(msg);
+      }
```

Comment:
It’s unusual to reject with a string. Usually it’s an Error

Reply:
This is in silent mode so maybe we should actually resolve with the string?

Reply:
Currently the string is returned in silent mode instead of thrown

Reply:
I updated it to resolve on silent and reject with error when not in silent mode. This seems to be the closest async equivalent to what happens when synchronous.